client/ide/lib/realtimemanagerfactory.coffee

```diff
@@ -8,8 +8,363 @@ module.exports = class RealtimeManagerFactory extends KDObject
 
   get: (type) ->
       if type is 'FIREBASE'
-        realtime = new FirebaseRealtimeManager()
+        realtime = new FirebaseRealtimeManager
       else if type is 'GOOGLE_DRIVE'
-        realtime = new GoogleDriveRealtimeManager()
+        realtime = new GoogleDriveRealtimeManager
       return realtime
   
+  setRealtimeDoc: (realtimeDoc) ->
+
+    @realtimeDoc = realtimeDoc
+    
+  getRealtimeDoc: ->
+
+    unless @realtimeDoc
+      throw new Error 'RealtimeDoc is not set yet for FirebaseRealtimeManager'
+
+    return @realtimeDoc
+    
+  createFile: (options, callback) ->
+
+    { title, preventEvent } = options
+
+    return throw new Error 'title is required'  unless title
+
+    options      =
+      resource   :
+        mimeType : 'application/vnd.google-apps.drive-sdk'
+        title    : title
+    if type is 'FIREBASE'
+      firebase.put(options).execute (file) =>
+      callback null, file
+    else if type is 'GOOGLE_DRIVE'
```

Comment:
Factory class is not an interface type. It doesn't need to declare methods.

You should've created a new class named `RealtimeManagerInterface`. `RealtimeManagerFactory` should have only a getter instantiating new instances.